test/common/formatter/substitution_format_string_test.cc

```diff
@@ -135,5 +155,71 @@ TEST_F(SubstitutionFormatStringUtilsTest, TestFromProtoConfigFormatterExtensionU
                             "Formatter not found: envoy.formatter.TestFormatterUnknown");
 }
 
+TEST_F(SubstitutionFormatStringUtilsTest, TestFromProtoConfigJsonWithExtension) {
+  TestCommandFactory factory;
+  Registry::InjectFactory<CommandParserFactory> command_register(factory);
+
+  const std::string yaml = R"EOF(
+  json_format:
+    text: "plain text %COMMAND_EXTENSION()%"
+    path: "%REQ(:path)% %COMMAND_EXTENSION()%"
```

Comment:
im not totally following the code - but im thinking this will change logging for end users - can we add some docs ?

Reply:
So this PR is just a follow-up for what was missed in #14512 -- do you think the docs added there cover this? I think we should be Ok...

Reply:
ok, cool - i was trying to understand what had changed by looking at the tests

i will create some basic logging docs and look through the formatting docs more thoroughly then.

aplogies for noise

Reply:
Thanks for checking! Do let me know if I can help with input on how the extensions work (for the logging docs).

Reply:
I'm implementing command extension currently at #15711, which might be the first one? I'd appreciate reviews/comments there if you have time to check!